### ISSUE TITLE
Unable to send email error

### ISSUE THREAD

**Issue by RedLucas** [Shippable/support#4084]
We are inexplicably receiving false negative because of a "Unable to send email" error when we are building a Drupal 8 instance. 

I've tried to find some kind of pattern with our code that could be causing this, and the conclusion without a doubt is that it is unrelated. I am able to get the tests to pass by rerunning the build over and over until it magically works.

It's terribly inefficient to do this, so any help with this would be appreciated. The link below is just one of the many instances of this error. 

- https://app.shippable.com/github/advisorwebsites/d8platform/runs/13314/1/console

**Comment by ambarish2012** [comment]
@RedLucas is sending email to a specific address highly desirable in your build? I am not a drupal expert, but it seems to be an issue with the smtp module based on several posts i read about other uses running into this issue.

As per this post, https://drupal.stackexchange.com/questions/38255/how-can-i-stop-drupal-from-sending-e-mails-of-any-kind, there is an option to redirect emails to a file log rather than sending them to a recipient, which might help.

**Comment by RedLucas** [comment]
Ya, I mean I don't really understand why this is suddenly, and sporatically, an issue. It never used to be until later last week. Ideally I don't need to change any code on my application to make it work in shippable... but I'll try to look up that type of solution.

**Comment by RedLucas** [comment]
So is there no solution to this issue? I'd even like to revert whatever change on Shippable's end started this problem. 

It's not every test that fails, probably 1 in 5... always seems to be the most important build to, the one I'm watching..

**Comment by trriplejay** [comment]
Unfortunately we're not really seeing any change on our end that could have caused this problem.  It looks like the first time you experienced this was on 1/15, correct? we had official releases on 1/19 and before that 1/11.  The 1/11 release mostly dealt with small bugs and should not have had any impact on the build runtime of a CI job like this.

is it possible that some other dependency that you're using had an update or released a version that you automatically pull that could be causing this?

**Comment by RedLucas** [comment]
Ya, I'd been fighting with it for a few days before then. 

So, we have a line in our build that I believe previously fixed it, but perhaps it stopped being effective?

`    - echo "sendmail_path=$(which true)" >> $HOME/.phpenv/versions/$(phpenv version-name)/etc/php.ini` 

I'm really not sure, it's just really frustrating to get infrequent false negatives on our builds.

**Comment by a-murphy** [comment]
Was it an intermittent or consistent error before you added that line?    And, have you checked if anything had changed when it first appeared?

**Comment by RedLucas** [comment]
That line was added 6 months ago, when this issue had originally started. 

I thought that something had changed, since I had recently updated from Drupal 8.4.3 to 8.4.4. but the PR that updated it had passed. 

I still wanted to eliminate the possibility that it was related to that update, so I reverted all those changes and still had the problem. Plus, the problem appeared as well on the release that did not have this update. The problem also did not appear on the updated branch. 

I haven't been able to find any pattern as to why this is happening. I was almost hoping that like, the php.ini path in shippable has changed with an update and that broke it maybe? Could it _sometimes_ be different? (I donno, I'm just trying to help troubleshoot it here..)

**Comment by a-murphy** [comment]
Looking at the history of your project, that line was present over a long period of time and when your builds were running on three different images.  It's very unlikely that the error could have been caused by any changes in Shippable, and we definitely aren't changing the `php.ini` file in the middle of a build.  The path you are using appears to be correct, and the `php -i` command is finding the values you set, so unless you have another command that un-sets it, it should be correctly set.

Here are some things you can try, if you haven't already:
1. Clear the build cache.  A cached file can cause problems if it conflicts with a newer version of something.  There's a "reset cache" button near the bottom of the project settings page that will clear the cache once.  Because it's intermittent, you may also want to consider removing caching from your YML for a few days to see if it makes a difference.
2. Switch to the latest build image.  If you haven't specified the image in your shippable.yml, you can do this by [changing your subscription machine image](http://docs.shippable.com/platform/runtime/machine-image/ami-overview/#changing-the-subscription-machine-image).  If it's in your shippable.yml, you'll need to update the image tag there, too.
3. Increase `memory_limit`.  I don't know much about Drupal, but some other errors related to PHP have been resolved by increasing the memory limit.

**Comment by RedLucas** [comment]
We're using drush, and I found a workaround with these two parameters at the end of the `drush site-install` command `install_configure_form.enable_update_status_module=NULL install_configure_form.enable_update_status_emails=NULL`, Which avoids the issue in a different way. 

Still not sure what caused the issue to begin with, but an end to the false negatives are a win my any means. 

I also changed to a newer build image as well, but that's more just to keep things up to date at this point. 

Thanks for your help.
